### PR TITLE
feat: Add event id to outgoing messages

### PR DESCRIPTION
**Pull request by ebbeknudsen**
## Description
Adds EventId column to OutgoingMessages, which holds the id of the ServiceBusMessage's MessageId that triggerede the outgoing message. This makes it easier to track and debug the application.

## References
https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/176

### CODE REVIEW

File: source/BuildingBlocks.Domain/Models/EventId.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using System;
+using System.Text.Json.Serialization;
+
+namespace Energinet.DataHub.EDI.BuildingBlocks.Domain.Models;
+
+/// <summary>
+/// An EventId symbolises a tracking id, typically received from a ServiceBusMessage's MessageId,
+///     or in case of our IntegrationEvents, this is also the EventIdentifier
+/// </summary>
+public record EventId
```
Comment (RasmusGraabaek) [inline source/BuildingBlocks.Domain/Models/EventId.cs:24]:
```suggestion
[Serializable]
public record EventId
```

File: source/IntegrationTests/Application/Transactions/AggregatedMeasureData/AggregatedTimeSeriesRequestAcceptedToAggregationResultTests.cs [hunk 2]
```diff
@@ -394,7 +394,7 @@ await _gridAreaBuilder
 
         await _inboxEventReceiver
             .ReceiveAsync(
-                Guid.NewGuid().ToString(),
+                EventId.From(Guid.NewGuid()),
                 nameof(AggregatedTimeSeriesRequestAccepted),
                 process.ProcessId.Id,
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/Application/Transactions/AggregatedMeasureData/AggregatedTimeSeriesRequestAcceptedToAggregationResultTests.cs:399]:
```suggestion
```

File: source/IntegrationTests/TestBase.cs [hunk 3]
```diff
@@ -225,11 +226,11 @@ protected Task CreateActorIfNotExistAsync(CreateActorDto createActorDto)
             return GetService<IMasterDataClient>().CreateActorIfNotExistAsync(createActorDto, CancellationToken.None);
         }
 
-        protected async Task HavingReceivedInboxEventAsync(string eventType, IMessage eventPayload, Guid processId)
+        protected async Task HavingReceivedInboxEventAsync(string eventType, IMessage eventPayload, Guid processId, string? eventId = null)
         {
             await GetService<IInboxEventReceiver>().
                 ReceiveAsync(
-                    Guid.NewGuid().ToString(),
+                    EventId.From(eventId ?? Guid.NewGuid().ToString()),
```
Comment (RasmusGraabaek) [inline source/IntegrationTests/TestBase.cs:233]:
```suggestion
                    EventId.From(eventId ?? Guid.NewGuid()),
```

**Review by MadsDue — DISMISSED** [review]

File: source/IntegrationTests/TestBase.cs (hunk 3 above)
Comment (ebbeknudsen) [reply source/IntegrationTests/TestBase.cs:233]: Sadly the left hand side of the ?? is a string, so the right hand side has to return as string as well

**Review by MadsDue — APPROVED** [review]

**Pull request merged**